app/master/build_request_handler.py

```diff
@@ -107,110 +100,13 @@ def _prepare_build_async(self, build, project_lock):
             analytics.record_event(analytics.BUILD_PREPARE_START, build_id=build.build_id(),
                                    log_msg='Build preparation loop is handling request for build {build_id}.')
             try:
-                self._prepare_build(build)
+                build.prepare(self._subjob_calculator)
```

Comment:
Ah, yeah good question -- I considered that briefly (just instantiating this inside the Build object), but kind of waffled on it.

It's generally easier to test with injected dependencies, but with Python specifically it's not too hard to mock out even inline instantiations, so it's probably not a good enough reason for us to go adding a bunch of extra parameters to our constructors and methods. (I could see a stronger argument for dependency injection to support us creating a user plugin architecture, but that's not really on the roadmap.)

Mostly I did this just because the subjob calculator really only had to be instantiated once. Actually now the more I talk about it, the more I lean towards just hiding it from the build request handler and just instantiating it for each build. You're right that it is a bit of a break from our existing style. I might change this after all. :sunglasses: